common/extensions/api/brave_shields.json

```diff
@@ -49,6 +49,24 @@
             "parameters": []
           }
         ]
+      },
+      {
+        "name": "dappAvailable",
```

Comment:
this is a request to enable the wallet so I don't think `dappAvailable` makes sense as the name. Maybe something like `enableWallet`?

Reply:
also why is this in brave_shields? It doesn't appear to have anything to do with shields

Reply:
With some discussion with @bsclifton, I reused existing shields extension for reducing resources. Adding new extension consumes memory, cpu and slow down startup time.

Reply:
brave_shields api is not the same thing as the brave extension. A new extension api does not consume resources

Reply:
it's also weird that this api is available even when `BRAVE_WALLET_ENABLED` is not enabled. I think it would be better to create a new API and exclude everything in gn. It doesn't make any sense to have an api with no implementation https://github.com/brave/brave-core/pull/2678/files#diff-79f58b4f74941b52ddc06f0575de8c6bR91 
That's just confusing

Reply:
Ah right. sorry. I mis-understood your comment.

Reply:
I moved this to `brave_wallet.json` and only include it when `brave_rewards_enabled`